### DISCUSSION TITLE
How to Update collection in Milvus?

### DISCUSSION THREAD

**Discussion by NavaneethanL** [milvus-io/milvus#24258] [Q&A and General discussion]
I have a collection in milvus, where I uploaded my vectors generated from directory of files. now if I upload a new file to the directory how can I update my milvus collection with new set of vectors?

**Comment by yhmo** [comment]
To update a vector in a collection, use delete()+insert().
Assume you created a collection with two fields: "id" and "vec".
You have inserted some vectors with id=1, 2, 3, 4 ...
```
    data = [
        [1, 2, 3, 4],
        [[v1], [v2], [v3], [v4]],
    ]
    collection.insert(data)
```

Now you want to update the vector which id=3, you can:
`collection.delete(expr="id==3")`

Then insert the new vector:
```
    data = [
        [3],
        [[v3_new]],
    ]
    collection.insert(data)
```
